### PR TITLE
C++ codegen for reporting arrow data type for structs

### PR DESCRIPTION
**Pull request by Wumpf**
* Part of #2647
* Next step after #2707

### What

Implements `to_arrow_data_type` (same exists in Rust) for all C++ components and datatypes.

Started splitting up `cpp.rs` a tiny little bit, not gonna go the single file way `rust.rs` went there, too hard to handle now that there's more than one author on it.

Test it yourself with `cargo codegen && ./examples/cpp/minimal/build_and_run.sh`

Rough next steps:
* implement same for unions
* implement extension type decl (missing in this PR!), needed for instance to declare component types
* serialize out data
* testing (nothing in this PR is tested other than "it compiles!")
* utilities for better usability

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2756) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2756)
- [Docs preview](https://rerun.io/preview/pr%3Aandreas%2Fcpp-arrow-datatype-definition-for-structs/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aandreas%2Fcpp-arrow-datatype-definition-for-structs/examples)

### CODE REVIEW

File: crates/re_types_builder/src/codegen/cpp/mod.rs [hunk 1]
```diff
@@ -811,3 +825,103 @@ fn quote_docstrings(docs: &Docs) -> TokenStream {
...
+            DataType::Int64 => quote!(arrow::int64()),
+            DataType::UInt8 => quote!(arrow::uint8()),
+            DataType::UInt16 => quote!(arrow::uint16()),
+            DataType::UInt32 => quote!(arrow::uint32()),
+            DataType::UInt64 => quote!(arrow::uint64()),
+            DataType::Float16 => quote!(arrow::float16()),
+            DataType::Float32 => quote!(arrow::float32()),
+            DataType::Float64 => quote!(arrow::float64()),
+            DataType::Binary => quote!(arrow::binary()),
+            DataType::LargeBinary => quote!(arrow::::large_binary()),
```
Comment (teh-cmc) [inline crates/re_types_builder/src/codegen/cpp/mod.rs:856]: that's a lot of columns :flushed:

**Review by teh-cmc — APPROVED** [review]
LGTM.

It'd be nice to _not_ inline recursive datatypes into their parent but rather insert a runtime call to the the child's `to_arrow_datatype()` method to avoid code bloat and unreadable giant `to_arrow_datatype()` methods.

I need to do the same in Rust.

**Comment by Wumpf** [comment]
makes sense, gonna do a follow-up this morning with that!

File: crates/re_types_builder/src/codegen/cpp/mod.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_types_builder/src/codegen/cpp/mod.rs:856]: oops

**Pull request merged**